docs/tutorials/using_bdk_with_hardware_wallets.md

```diff
@@ -0,0 +1,217 @@
+Python 3.9.13 (main, May 17 2022, 14:19:07) 
+[GCC 11.3.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import hwi
+```
+
+If nothing happens, you're set! Instead, if you get a `ModuleNotFoundError`, follow the instructions in [HWI's README.md](https://github.com/bitcoin-core/HWI#bitcoin-hardware-wallet-interface) for installing.
+
+Warning: if you're using MacOS and `virtualenv`, you *may* encounter some problems with `rust-hwi`, as we internally use `PyO3`: https://github.com/PyO3/pyo3/issues/1741
+
+## Initial setup
+
+Start by creating a new Rust project:
+```
+$ cargo init bdk-hwi
+     Created binary (application) package
+$ cd bdk-hwi
+```
+
+Add `bdk` with the `hardware-signer` feature as a dependency in the `Cargo.toml`:
+```
+[package]
+name = "bdk-hwi"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+bdk = { version = "0.24.0", features = [ "hardware-signer", ] }
+```
+
+(`bdk` re-exports `rust-hwi` since version `0.24.0` - if you're using `bdk` <= `0.23.0`, you have to separately decleare `rust-hwi` as a dependency)
+
+Now, open `src/main.rs` and slightly modify the `fn main()` method to return a `Result`:
+```
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    println!("Hello, world!");
+    Ok(())
+}
+```
+This will come in handy later, as it allows us to use `?` on errors.
+
+Build and run the project - if everything went smoothly it will print a "Hello, world!" on the screen.
+```
+$ cargo run
+```
+
+## Finding the hardware wallet
+
+In this step we'll make sure that `hwi` can see your hardware wallet. If you're using a physical HW, connect it to your laptop; if it's an emulator, start it.
+
+We start by printing all the available hardware wallets:
+
+```
+use bdk::hwi::HWIClient;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Listing all the available hardware wallet devices...
+    let devices = HWIClient::enumerate()?;
+    println!("{:?}", &devices);
+    Ok(())
+}
+```
+
+When run, it should print an array of `HWIDevice` with one element:
+
+```
+$ cargo run
+[HWIDevice { ... }]
+```
+
+If the array is empty instead, `hwi` is having troubles recognizing your device. Common issues are: the device is locked (unlock with the pin and open the "Bitcoin" app, if needed) or the udev rules aren't set.
+
+## Receiving funds
+
+In order to be able to receive funds we need to create the BDK `Wallet` using the HW descriptors.
+
+We start by creating a `HWIClient` from the `HWIDevice` we found in the last step:
+
+```
+// Listing all the available hardware wallet devices...
+let devices = HWIClient::enumerate()?;
+let first_device = devices
+    .first()
+    .expect("No devices found. Either plug in a hardware wallet, or start a simulator.");
+// ...and creating a client out of the first one
+let client = HWIClient::get_client(first_device, true, HWIChain::Test)?;
```

Comment:
I get an error on this step both on macOS and Linux. I'm not sure what It could be. 
Once tutorial is updated I can retry the tutorial. 

```
11 | let client = HWIClient::get_client(first_device, true, HWIChain::Test)?;
   |                                                        ^^^^^^^^ use of undeclared type `HWIChain
```

Reply:
Oh yeah, it's because you have to import `HWIChain`, I'll specify that *somewhere*

Reply:
I've added a section at the start of the post with all the imports needed :)